# This is the index into node_positions set
# to determine the node which sets the zero coordinate of the structure
offset_node_index: 0

# List of all nodes in the structure
# Values correspond to x,y,z in meters

node_positions:
  - position: [0.0, 0.0, 0.0]
  - position: [1000.0, 0.0, 0.0]
  - position: [2000.0, 0.0, 0.0]
  - position: [3000.0, 0.0, 0.0]
  - position: [4000.0, 0.0, 0.0]
  
# List of all beams in the structure
# List correspond to pairs of indices of nodes
# nodes: [0, 1] constructs a beam from node 0 to node 1
beams:
  - nodes: [0,1]
  - nodes: [1,2]
  - nodes: [2,3]
  - nodes: [3,4]

# Which beams are sensors
# nodes: [[2, 4], [1, 7]] constructs sensor from node 2 to node 4 and node 1 to 7
# both the sensors are connected 
sensors:
  -
    nodes: [[0,1]]
    teensy: 0

# Which nodes have light strips wrapped around
light_nodes: 
  -
    nodes: [0, 1, 2, 3, 4]
    num_leds: [2, 2, 2, 2, 2]
    num_leds_per_meter: 60
    teensy: 0

#block_light_beams:
#  -
#    nodes: [[10,28], [11,30], [19,40], [23,45]]
#    centers: [0.5, 0.5, 0.5, 0.5]
#    num_leds: [30, 30, 30, 30]
#    num_leds_per_meter: 30
#    teensy: 0

# Which beams have light strips
# nodes: [0, 1] constructs a light strip from node 0 to node 1
pixel_light_beams:
  -
    nodes: [[0, 1], [1, 2]]
    centers: [0.5, 0.5]
    num_leds: [30, 30]
    num_leds_per_meter: 30
    teensy: 0

# Size of the nodes and color
nodes_size:
  x: 0.1016
  y: 0.1016
  z: 0.1016
nodes_color: [0.8, 0.8, 0.8, 1.0] # r,g,b,a

# Size of the light nodes and color
block_nodes_size:
  x: 0.13
  y: 0.13
  z: 0.13
block_nodes_color: [0.6, 0.6, 0.6, 1.0] # r,g,b,a

# Size of the beams and color
beams_size:
  x: 0.0317
  y: 0.0317
  z: 1.4 
beams_color: [0.9, 0.9, 0.9, 0.7] # r,g,b,a

# Size of the block light beams and color
block_beams_size:
  x: 0.04
  y: 0.04
  z: 0.9 
block_beams_color: [0.8, 0.4, 0.4, 1.0] # r,g,b,a

# Size of the pixel light beams and color
pixel_beams_size:
  x: 0.04
  y: 0.04
  z: 0.9
pixel_beams_color: [0.8, 0.4, 0.4, 1.0] # r,g,b,a

# Size of the sensors and color
sensors_size:
  x: 0.04
  y: 0.04
  z: 1.4
sensors_color: [0.9, 0.9, 0.9, 0.4] # r,g,b,a

